### DISCUSSION TITLE
A question about the Frame.Navigate

### DISCUSSION THREAD

**Discussion by Ac-Ada-Lovelace** [microsoft/WinUI-Gallery#1598] [Q&A]
I have a question regarding Microsoft's WinUI Gallery implementation. In the `NavigationRootPage.xaml.cs` file, there is a method defined as follows:

```csharp
public void Navigate(
    Type pageType,
    object targetPageArguments = null,
    Microsoft.UI.Xaml.Media.Animation.NavigationTransitionInfo navigationTransitionInfo = null)
{
    NavigationRootPageArgs args = new NavigationRootPageArgs();
    args.NavigationRootPage = this;
    args.Parameter = targetPageArguments;
    rootFrame.Navigate(pageType, args, navigationTransitionInfo);
}
```

The `rootFrame.Navigate(pageType, args, navigationTransitionInfo);` line uses the variable `args`, which is of type `NavigationRootPageArgs`. The definition of `NavigationRootPageArgs` is as follows:

```csharp
public class NavigationRootPageArgs
{
    public NavigationRootPage NavigationRootPage;
    public object Parameter;
}
```

According to Microsoft's official documentation, the `parameter` for the `Navigate` method is described as:

```
parameter
Object
The navigation parameter to pass to the target page; must have a basic type (string, char, numeric, or GUID) to support parameter serialization using GetNavigationState.
```

Given this, how can this definition work with the `Navigate` method? Additionally, are there any other workable implementations for this scenario?

I consulted ChatGPT and looked through some similar code on GitHub, but I haven't found anything helpful so far. I was expecting to understand how the NavigationRootPageArgs type can work with the Navigate method despite the documentation stating that the parameter should have a basic type (string, char, numeric, or GUID) to support parameter serialization using GetNavigationState.

**Comment by marcelwgn** [comment]
I see where the confusion comes from. Essentially what the documentation is saying, if you want to call `Frame.GetNavigationState`, you must provide one of the basic types.
If you don't use one of these types, calling `GetNavigationState` actually throws an exception. However, if you don't need to call this method, you can pass in anything you like which the WinUI Gallery is doing here. 

On a side note, apparently you shouldn't form a dependency on the serialization mechanism for `GetNavigationState`: https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.frame.getnavigationstate?view=winrt-26100#remarks
